frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx

```diff
@@ -2,13 +2,29 @@ import * as React from 'react';
 import PerspectiveDetector from './PerspectiveDetector';
 import { PerpsectiveContext } from './perspective-context';
 import { useValuesForPerspectiveContext } from './useValuesForPerspectiveContext';
+import { isPerspective, Perspective, useExtensions } from '@console/plugin-sdk';
 
 type DetectPerspectiveProps = {
   children: React.ReactNode;
 };
 
+const getPerspectiveURLParam = (perspectives: Perspective[]) => {
+  const perspectiveIDs = perspectives.map(
+    (nextPerspective: Perspective) => nextPerspective.properties.id,
+  );
+
+  const urlParams = new URLSearchParams(window.location.search);
+  const perspectiveParam = urlParams.get('perspective');
+  return perspectiveIDs.includes(perspectiveParam) ? perspectiveParam : null;
+};
+
 const DetectPerspective: React.FC<DetectPerspectiveProps> = ({ children }) => {
   const [activePerspective, setActivePerspective, loaded] = useValuesForPerspectiveContext();
+  const perspectiveExtensions = useExtensions<Perspective>(isPerspective);
+  const perspectiveParam = getPerspectiveURLParam(perspectiveExtensions);
+  if (perspectiveParam && perspectiveParam !== activePerspective) {
+    setActivePerspective(perspectiveParam);
+  }
```

Comment:
Redux actions should not be executed inside of the render function. Use `useEffect`

Reply:
fixed, thanks